### DISCUSSION TITLE
Automatic Factory Bot associations

### DISCUSSION THREAD

**Discussion by Splines** [shakacode/cypress-playwright-on-rails#166] [General]
First of all, thanks for this neat library acting as interface between cypress and Rails!

From your [factory bot association docs](https://github.com/shakacode/cypress-on-rails/blob/master/docs/factory_bot_associations.md):

> You cannot access associations directly from Cypress like you can do with Ruby tests. So setting up associations has to be done differently from within Cypress.

All these methods feel kind of "hacky". When I specify an [`assocation`](https://thoughtbot.github.io/factory_bot/ref/association.html) command in factory bot, it automatically works in Ruby on Rails RSpec tests when calling `FactoryBot.create(...)`. I'm wondering why we can't have this reflected in the cypress `appFactories` command and instead have to set this up manually from Cypress?

In the end, the code is executed by `FactoryBot` via Ruby on Rails, isn't it? So in theory, shouldn't it be possible to also deal with associations in the backend? Or maybe you can elaborate on why this design decision was chosen such that it has to be done manually?

**Comment by grantspeelman** [comment]
this is because we actual making a web request to the backend, and thus can only send JSON and don't have access to the actual ruby objects in cypress.

**Reply by Splines** [reply]
Ok, so I definitely wanted to have this feature and just implemented it [here](https://github.com/Splines/cypress-rails-factory-bot), feel free to draw inspiration from it ;) I'm now able to call FactoryBot from Cypress code like this:

```js
FactoryBot.create("assignment", "with_deadline_tomorrow", { file_type: ".pdf", size_max: 10 });
```

**Reply by grantspeelman** [reply]
I see, removing the brackets `[]` would make it look closer to ruby. This does seem easier to get started with 👍🏽

**Reply by Splines** [reply]
If by `[]` you mean `{}`, that's not possible as this wouldn't be valid JS anymore. The best equivalent I could find to Ruby hashes is an object in JS.

**Reply by grantspeelman** [reply]
what i mean is, the current code makes it look like this: which has the extra `[]`, but what you suggesting makes it look closer to how it's used in ruby

```js
cy.appFactories([
  ["create", "assignment", "with_deadline_tomorrow", { file_type: ".pdf", size_max: 10 }]
])
```

**Reply by Splines** [reply]
Ah yeah exactly. But not just that, another thing is that I provide a convenience wrapper, such that we can call `FactoryBot.create()` in Cypress tests and not have to pass the method name `"create"` as string to another function.
